### ISSUE TITLE
Errors during the update of lucene version (from 5.3.2 to 6.5.1)

### ISSUE THREAD

**Issue by MargotPiva** [openwide-java/owsi-core-parent#78]
The class NumericRangeQuery (https://lucene.apache.org/core/5_2_1/core/org/apache/lucene/search/NumericRangeQuery.html) no longer exists which caused errors in LuceneUtils.
Instead of this class, we should use the PointRangeQuery class but it does not implement the inclusion/exclusion of the upper/lower bound.

As a temporary fix we have implemented a simplified version of NumericRangeQuery that extends Query instead of MultiTermQuery. We have changed the package on LuceneUtils and tested this new class in TestLuceneUtils.

NumericRangeQuery is marked as deprecated and will be deleted when we make the transition with PointRangeQuery.

**Comment by yrodiere** [comment]
Hello,

For your information, Hibernate Search isn't compatible with Lucene 6.0 yet, and by this I mean we didn't even look at what would be required to migrate it. You may get in a lot of trouble by attempting to make this work.
Of course, if you are moving away from Hibernate Search, my comment is not relevant.

Cheers.

**Comment by gsmet** [comment]
Yeah, the idea would be to upgrade the main components e.g. Spring, Hibernate, Hibernate Search... and be consistent regarding the dependencies.

Typically, it's Hibernate Search that drives the version of Lucene used in the core.

**Comment by lalmeras** [comment]
Hi,

Effectively we make a mistake when we choose target versions to update dependencies. I agree not to select Lucene 6.0 as target version, and that Hibernate Search's requirements should be used to select Lucene's version.

@MargotPiva please cancel your lucene's version update and do it consistently with Hibernate Search requirements.
@gsmet @yrodiere thanks for your warning.

Thanks.
